src/sentry/services/hybrid_cloud/app/__init__.py

```diff
@@ -20,10 +21,13 @@ def find_installation_by_proxy_user(
         pass
 
     @abc.abstractmethod
-    def get_installed_for_organization(
+    def get_many(
         self,
         *,
-        organization_id: int,
+        organization_id: Optional[int] = None,
```

Comment:
Be careful when we create wide query interfaces like this -- I went ahead and modified the impl to return an `[]` when neither organization_id or api_token_id are given.  A bug that might provide a None value to either of these parameters, as written, would result in returning every application.

Reply:
This is especially true anytime we go from non filtered results to conditionally filtered based on None.  Python won't distinguish "None because this was an optional parameter" and "None because the caller accidentally didn't provide a value".

In some ways... I prefer not having these wide query interfaces like `get_many`.  `get_installed_for_organization` was way more clear; if I provide a None, it matches on None, returning 0 results, as opposed to ignoring the filtering for organization.